pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -1106,12 +1110,39 @@ func (c InstallerController) ensureRequiredResourcesExist(ctx context.Context, r
 }
 
 func (c InstallerController) Sync(ctx context.Context, syncCtx factory.SyncContext) error {
-	operatorSpec, originalOperatorStatus, _, err := c.operatorClient.GetStaticPodOperatorState()
+	operatorSpec, originalOperatorStatus, operatorResourceVersion, err := c.operatorClient.GetStaticPodOperatorState()
 	if err != nil {
 		return err
 	}
 	operatorStatus := originalOperatorStatus.DeepCopy()
 
+	// Perform a live get to obtain the RV of the object the controller applied
+	// because for the MOM effort it is not allowed to get the RV from
+	// Apply response.
+	if c.podOperatorStatusApplied {
+		_, _, resourceVersion, err := c.operatorClient.GetOperatorStateWithQuorum(ctx)
```

Comment:
Would it be more robust to save the RV from the first live get?

As written, this controller will only proceed when its cached RV is exactly the same as the RV from a live get. This requires at least 3 network hops: apiserver sends watch event, operator sends live get request, server sends live get response. Imagine another client is continuously spamming writes to the object more often than the time it takes for those 3 hops to complete.

Reply:
oh yes definitely, I just moved code around with the new change, but it would definitely be better that way